packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -524,6 +592,29 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
         silent: true,
       );
     });
+    if (chunkEvents != null) {
+      chunkEvents.listen(
+        (ImageChunkEvent event) {
+          if (hasListeners) {
+            final List<ImageChunkListener> localListeners = _listeners
+                .map<ImageChunkListener>((ImageStreamListener listener) => listener.onChunk)
+                .where((ImageChunkListener chunkListener) => chunkListener != null)
+                .toList();
```

Comment:
Why do you need this to be a list? The for loop below could just iterate over an iterable as well?

Reply:
To make a copy to allow for concurrent modification.

Reply:
Makes sense. Maybe add a short comment?

Reply:
Done